ir/src/main/scala/org/scalajs/core/ir/Serializers.scala

```diff
@@ -634,11 +646,36 @@ object Serializers {
     }
 
     def readTree(): Tree = {
+      val pos = readPosition()
+      readTreeFromTag(input.readByte())(pos)
+    }
+
+    def readOptTree(): Option[Tree] = {
+      if (useHacks068) {
+        readTreeAllowEmpty()
+      } else {
+        import input._
+        if (readBoolean()) {
+          Some(readTree())
+        } else {
+          None
+        }
+      }
+    }
+
+    private def readTreeAllowEmpty(): Option[Tree] = {
+      require(useHacks068)
+      val pos = readPosition()
+      val tag = input.readByte()
+      if (tag == TagEmptyTree) None
+      else Some(readTreeFromTag(tag)(pos))
```

Comment:
What about the position? The problem with reading a tree, is that we need to read a position first. We could (should?) inverse tree tags and positions, but I'm not sure this is a good thing to do in a minor IR version (although possible).